main.go

```diff
@@ -291,6 +290,17 @@ func main() {
 		LeaderElectionNamespace: telemetryNamespace,
 		LeaderElectionID:        "cdd7ef0b.kyma-project.io",
 		CertDir:                 certDir,
+		NewCache: cache.BuilderWithOptions(cache.Options{
+			SelectorsByObject: cache.SelectorsByObject{
+				&corev1.Secret{}:         {},
+				&appsv1.Deployment{}:     {Field: fields.SelectorFromSet(fields.Set{"metadata.namespace": telemetryNamespace})},
```

Comment:
Could you please extract `fields.SelectorFromSet(fields.Set{"metadata.namespace": telemetryNamespace})` into a helper function?